test/html/test_html.py

```diff
@@ -699,3 +699,11 @@ def test_html_ol_ul_line_height(tmp_path):
     </ul>"""
     )
     assert_pdf_equal(pdf, HERE / "html_ol_ul_line_height.pdf", tmp_path)
+
+
+def test_html_long_list_entries(tmp_path):
```

Comment:
You have added functionality to text_region.py, which is very useful even when not parsing HTML. That means we need tests to verify that the new arguments for `Paragraph()` work correctly when used directly as well. This is necessary to avoid regressions with any future changes.

It is also a good reason to avoid changing line_break.py, because otherwise you'd have to add tests to verify your changes there as well... :wink:

Reply:
Will do.

Reply:
On second thought, I would like to also ask what tests I should make for that.

Reply:
Those tests should allow to verify that `Paragraph`s are rendered correctly with any combination of indent and bullet string either present or not, both with bullet strings that are longer or shorter than the text indent is wide.

Reply:
I have added a test for long `<ol>` bullets. The other tests that are already present seem to already fulfil the need for testing whether or not a `Paragraph` is rendered correctly with different `bullet`s and `tag_indents`. Am I missing something?

Reply:
> Am I missing something?

You're missing the simple fact that bulleted/numbered paragraphs can be used outside of parsing HTML (went in more detail with other comments).

Reply:
> You're missing the simple fact that bulleted/numbered paragraphs can be used outside of parsing HTML (went in more detail with other comments).

I currently don't see `Paragraph` being used outside of handling HTML, hence why the assumption.

> The new functionality also must be added to the documentation of the .paragraph() method (both in the docstring and in "doc/TextRegion.md")

There don't seem to be docstrings for `ParagraphCollectorMixin.paragraph()` and `Paragraph`. Should I create one?

Reply:
> I currently don't see Paragraph being used outside of handling HTML, hence why the assumption.

Explained above.

> There don't seem to be docstrings for ParagraphCollectorMixin.paragraph() and Paragraph. Should I create one?

That would be very helpful! It looks like I have been skimping a bit on docstrings in that module.

Reply:
Added the docstring to the `Paragraph` class. `TextRegion.md` has been edited. Test for generation of `Paragraph` objects has been added, albeit it does not test the handling of `bullet_rel_x_displacement` and `bullet_rel_y_displacement`, as relevant `Paragraph`-instantiating methods do not have the relevant functionality yet.

I am considering removing `bullet_rel_y_displacement` from `Paragraph` altogether in favour of vertical alignment, but the introduction of alignment will require some additional time.

Reply:
I'd recommend to leave vertical bullet alignment as a future enhancement. Let's get this stable and robust first, and then think about any more bells and whistles.

Reply:
Bumping.
A commit is ready for examination.

Reply:
Apologies for bumping again.  
Is there anything else required to do regarding this?